### PR TITLE
Add search test written in Python

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/91257

Verification run: https://openqa.opensuse.org/t1840023

### CODE REVIEW

**Review by grisu48 — COMMENTED** [review]
This looks already very promising!

**Comment by okurz** [comment]
actually good idea :)

File: tests/openQA/search.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    assert_screen('openqa-logged-in')
+    assert_and_click('openqa-search')
+    type_string('shutdown.pm')
+    send_key('ret')
+    assert_screen('openqa-search-results')
+
+
+def post_fail_hook(self):
+    send_key('ctrl-alt-f3')  # switch to root console
+    script_run('openqa-cli api --o3 experimental/search q=shutdown.pm')
```
Comment (okurz) [inline tests/openQA/search.py:29]: as you run within the openQA-in-openQA instance that setup an openQA instance I recommend to connect to that, not "o3". As "localhost" is default I suggest to simply delete the parameter and then have a test that works with that instance.

File: tests/openQA/search.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    save_screenshot()
```
Comment (okurz) [inline tests/openQA/search.py:30]: Isn't there a screenshot already after the script_run call finished?

File: tests/openQA/search.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    assert_screen('openqa-search-results')
+
+
+def post_fail_hook(self):
+    send_key('ctrl-alt-f3')  # switch to root console
+    script_run('openqa-cli api experimental/search q=shutdown.pm')
+
+
+def test_flags(self):
+    return dict([('fatal', 1)])
```
Comment (Martchus) [inline tests/openQA/search.py:33]:
```suggestion
    return {'fatal': 1}
```

File: main.pm [hunk 4]
```diff
@@ -59,6 +59,11 @@ sub load_openQA_tests() {
     }
 }
 
+sub load_python_tests() {
+    # use Inline Python => "import os.path, sys; sys.path.insert(0, os.path.abspath(os.path.join(os.path.curdir, '../..')))";
```
Comment (okurz) [inline main.pm:63]: delete dead code

File: tests/openQA/search.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
```
Comment (okurz) [inline tests/openQA/search.py:15]: Why do you want to introduce a GPL licensed file? Other files within the repo don't have that license header and the project is MIT-licensed. I suggest to simply delete the header

File: tests/openQA/search.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+def run(self):
+    assert_screen('openqa-logged-in')
+    assert_and_click('openqa-search')
+    type_string('shutdown.pm')
+    send_key('ret')
+    assert_screen('openqa-search-results')
+
+
+def post_fail_hook(self):
+    send_key('ctrl-alt-f3')  # switch to root console
```
Comment (okurz) [inline tests/openQA/search.py:28]:
```suggestion

def switch_to_root_console():
    send_key('ctrl-alt-f3')

def post_fail_hook(self):
```

File: tests/openQA/search.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    script_run('openqa-cli api experimental/search q=shutdown.pm')
```
Comment (okurz) [inline tests/openQA/search.py:29]: Please use `assert_script_run` unless you want to check the exit code explicitly

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by okurz — APPROVED** [review]
This looks great now and could act as a very good example!

Care to mention it in the openQA documentation as showcase example? :)

**Pull request merged**